breathecode/feedback/tests/urls/tests_academy_survey.py

```diff
@@ -0,0 +1,490 @@
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_academy_survey__post__without_academy(self):
+        """Test /academy/survey without authorization"""
+        self.bc.database.create(authenticate=True)
+        url = reverse_lazy('feedback:academy_survey')
+        response = self.client.post(url, {})
+        json = response.json()
+        expected = {
+            'detail': "Missing academy_id parameter expected for the endpoint url or 'Academy' header",
+            'status_code': 403
+        }
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_academy_survey__post__without_role(self):
+        """Test /academy/survey without authorization"""
+        self.headers(academy=1)
+        url = reverse_lazy('feedback:academy_survey')
+        self.bc.database.create(authenticate=True)
+        response = self.client.post(url, {})
+        json = response.json()
+        expected = {
+            'detail': "You (user: 1) don't have this capability: crud_survey for academy 1",
+            'status_code': 403
+        }
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_academy_survey__post__without_cohort(self):
+        """Test /academy/survey post without data"""
+
+        self.headers(academy=1)
+        model = self.bc.database.create(
+            authenticate=True,
+            academy=True,
+            profile_academy=True,
+            role='crud_survey',
+            capability='crud_survey',
+        )
+
+        url = reverse_lazy('feedback:academy_survey')
+        response = self.client.post(url)
+        json = response.json()
+        expected = {'cohort': ['This field is required.']}
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_academy_survey__post__without_teacher_assigned(self):
```

Comment:
Test the exception in the serializer`'Minimum duration for surveys is one hour'`

Reply:
Remember add the slug in all the `ValidationException`